### ISSUE TITLE
parser: module information does not include type, interface, or namespace exports

### ISSUE THREAD

**Issue by nebrius** [oxc-project/oxc#10556]
This one is something of a follow-up to https://github.com/oxc-project/oxc/issues/10505#issuecomment-2818961656 for some other instances where types are not listed in module outputs (sorry for not thinking to check this edge case when I filed the first bug).

```js
import oxc from 'oxc-parser';

const code = `export type Foo = {}

export interface Bar {}

export namespace Baz {}`;

const result = oxc.parseSync('/project/a.ts', code, {
  sourceType: 'module',
});

console.log(result.module.staticExports);
```

Actual output:

```
[] // empty array
```

Expected output:

An array with three entries describing the three type exports.

**Comment by Boshen** [comment]
I didn't anticipate for TypeScript exports so they were excluded.

**Comment by nebrius** [comment]
> I didn't anticipate for usages of TypeScript exports so they were excluded.

No worries, I fully relate to the whack-a-mole nature of this kind of work. I've been doing various forms of code analysis based on TypeScript ASTs for many years, and I also still kinda regularly forget various edge cases 😃 

And again, the performance of OXC compared to typescript-eslint's parser is an absolute breath of fresh air!

**Comment by magic-akari** [comment]
May I take ownership of this implementation? I'd be happy to prepare a PR once completed.

**Comment by ulrichstark** [comment]
This issue is probably also closed by #10532

**Comment by nebrius** [comment]
I just tested this fix with a locally built copy of OXC, and it works perfectly. Thanks everyone for the fixes! Do y'all have a sense when the updated package will be published to npm?

**Comment by Boshen** [comment]
> I just tested this fix with a locally built copy of OXC, and it works perfectly. Thanks everyone for the fixes! Do y'all have a sense when the updated package will be published to npm?

Released in oxc v0.67.0
